rules/framework/vfs_overlay.bzl

```diff
@@ -79,7 +79,7 @@ def _build_subtrees(paths, vfs_prefix):
 
                 curr_subdirs.dict[part] = next_subdirs
                 curr_subdirs.json["contents"].append(next_subdirs_json)
-                curr_subdirs = next_subdirs
```

Comment:
@thiagohmcruz from what I understand this was a crux of the [original PR](https://github.com/bazel-ios/rules_ios/pull/492/files#diff-450b1330485549712d2fd4e65b2b3fd0a7692fe8f4ec43a0b7b60bbb739313afL80)

If this feature generally works but @tinder-maxwellelliott has the one case it's busted, perhaps it makes sense to leave their fix in for now, but behind a feature flag. Ideally we can get a test case for his issue or synthesize one our self.

Reply:
```
if ctx.attr.vfs_workaround_enabled:
    # do workaround 
else:
    # do original
```

There might be a non trivial upstream bug in the merging. This method could stand for a test case. The description of what is supposed to do is also pretty well defined, but riddled with edge cases

Reply:
Yep all that makes sense!

>makes sense to leave their fix in for now, but behind a feature flag

That's what I suggested in item (3) above in my latest comment. Definitely want to find a solution that works for all of us here while we investigate/try to come up with a repro for the original issue.

Reply:
See: https://github.com/bazel-ios/rules_ios/pull/504#issuecomment-1173980344